runtime/jvmti/jvmtiThread.c

```diff
@@ -67,51 +67,61 @@ jvmtiGetThreadState(jvmtiEnv* env,
 		} else {
 			threadObject = *(j9object_t *)thread;
 		}
-
-		/* get the lock for the object */
-		threadObjectLock = J9VMJAVALANGTHREAD_LOCK(currentThread,threadObject);
-
-		/* now get the vmThread for the object and whether the thread has been started, we get
-		 * these under the thread object lock
-		 * so that we get a consistent view of the two values */
-		if (threadObjectLock != NULL){
-			rc = getVMThread(currentThread, thread, &targetThread, TRUE, FALSE);
-			threadStartedFlag = (jboolean) J9VMJAVALANGTHREAD_STARTED(currentThread, threadObject);
-		} else {
-			/* in this case we must be early in the thread creation.  We still need to call getVMThread so that
-			 * the inspection count etc. are handled correctly, however, we just want to fall into the case were
-			 * we return that the thread is NEW so we set the targetThread and threadStartedFlag to false.
+#if JAVA_SPEC_VERSION >= 19
+		if (IS_VIRTUAL_THREAD(currentThread, threadObject)) {
+			/* If thread is NULL, the current thread is used which cannot be a virtual thread.
+			 * There is an assertion inside getVirtualThreadState() that thread is not NULL.
 			 */
-			rc = getVMThread(currentThread, thread, &targetThread, TRUE, FALSE);
-			targetThread = NULL;
-			threadStartedFlag = JNI_FALSE;
-		}
+			rv_thread_state = getVirtualThreadState(currentThread, thread);
+		} else {
+#endif /* JAVA_SPEC_VERSION >= 19 */
```

Comment:
With this change, the `ifdef` in the corresponding `}` can be avoided.
```suggestion
		} else
#endif /* JAVA_SPEC_VERSION >= 19 */
		{
```

Reply:
Updated.